doc/source/WorkingWithThePythonScriptingAPI/ThePythonEditor/index.md

```diff
@@ -0,0 +1,72 @@
+```eval_rst
+.. role:: raw-html(raw)
+    :format: html
+```
+
+# The Python Editor #
+
+The _Python Editor_ is a type of editor in the main Gaffer application that lets you build and modify the node graph, test expressions, and inspect data by inputting and executing Python code. In the default layout, it resides in the bottom-right panel, in a tab next to the _Hierarchy View_.
+
+The _Python Editor_ is split into two areas. The bottom-half is the input field. The top-half is the output. The input field of the _Python Editor_ functions like a basic plain text editor. You can type, select, cut, copy, paste, and execute code. 
+
+![A Python Editor with its input and the output field](images/interfacePythonEditorOutput.png "A Python Editor with its input and the output field")
+
+The output field behaves like a typical console output. It records the output stream, including return values and errors. Errors appear in red.
+
+![Errors in a Python Editor's output field](images/interfacePythonEditorError.png "Errors in a Python Editor's output field")
+
+> Tip :
+> When creating a new node using the _Python Editor_, you must first `import` the node's module, which will persist for the lifetime of the editor. However, you never need to `import` the [Gaffer](../../Reference/NodeReference/Gaffer/index.md) or [IECore](../../Reference/NodeReference/IECore/index.md) modules. They are loaded by default.
+
+
+## Instructions ##
+
+
+### Executing Code ###
+
+To execute code with the _Python Editor:_
+
+1. Enter code into the input field.
+2. Highlight any portion of the input you wish to preserve for reuse.
+3. Hit <kbd>Control</kbd> + <kbd>Enter</kbd>.
```

Comment:
> 2. Highlight any portion of the input you wish to preserve for reuse.

The behaviour is actually a little different to this:
 - If nothing is highlighted the entire input is evaluated then cleared.
 - If a any of the input is highlighted _only the highlighted code is evaluated_ and the entire input is preserved.